### PR TITLE
New issue: Normative permission to specialize Ranges variable templates

### PR DESCRIPTION
**Pull request by CaseyCarter**

### CODE REVIEW

**Comment by timsong-cpp** [comment]
We need to say that being `false` is not a requirement that user specializations need to meet.

File: xml/issue3183.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+<blockquote>
+<p>
+[&hellip;]
+</p><p>
+(2.2) &mdash; If <tt>−N</tt> is representable by
+<tt>iter_difference_t&lt;I&gt;</tt>, then <tt>i - s</tt> is well-defined and
+equals <tt>−N</tt>.
+</p><p>
+<ins>-?- Pursuant to <sref ref="[namespace.std]"/>, users may specialize
+<tt>disable_sized_sentinel</tt> for cv-unqualified types <tt>S</tt> and
```
Comment (timsong-cpp) [inline xml/issue3183.xml]: Why disallow cv-qualification but allow references?

**Comment by CaseyCarter** [comment]
> We need to say that being `false` is not a requirement that user specializations need to meet.

Technically we don't - [namespace.std] doesn't require program-defined specializations of standard library variable templates to meet any particular requirements. We *do* need to require that user specializations are boolean constant expressions.

File: xml/issue3183.xml (hunk 1 above)
Comment (CaseyCarter) [reply xml/issue3183.xml]: No good reason. Do you see any holes in "cv-unqualified non-array object types"?

Comment (timsong-cpp) [reply xml/issue3183.xml]: OK, but should we include pointers too? Do we want to allow specializing `disable_sized_sentinel<MyType*, MyType*>` to true?

Comment (CaseyCarter) [reply xml/issue3183.xml]: No, but we *do* want to allow `disable_sized_sentinel<int*, MySentinel>` and `disable_sized_sentinel<MyIterator, int*>`. 

"cv-unqualified non-array object types, at least one of which depends on a program-defined type, and at least one of which is not a pointer type"?

Comment (CaseyCarter) [reply xml/issue3183.xml]: "cv-unqualified non-array object types, at least one of which is a program-defined type"?

**Comment by CaseyCarter** [comment]
`enable_view` has normative permission to specialize, but also needs to require such specializations to be constant expressions of type `bool`.

**Comment by CaseyCarter** [comment]
This latest push addresses all earlier comments.

**Pull request merged**